lms/static/scripts/frontend_apps/services/grading.js

```diff
@@ -0,0 +1,66 @@
+import { apiCall } from '../utils/api';
+
+/**
+ * @typedef Student
+ * @prop {string} LISResultSourcedId - Unique outcome identifier
+ * @prop {string} LISOutcomeServiceUrl - API URL for posting outcome results
+ */
+
+/**
+ * The fetched grade success result.
+ *
+ * @typedef FetchGradeResult
+ * @prop {number} currentScore - The fetched grade
+ */
+
+/**
+ * Service for fetching and reporting student grades for an assignment.
+ */
+export class GradingService {
+  /**
+   * @param {object} options
+   *   @param {string} options.authToken
+   */
+  constructor({ authToken }) {
+    this._authToken = authToken;
```

Comment:
This is actually quite a good example of the value of the services layer. It has been bugging me how much `authToken` is swapped around between components!